sqlmesh/core/model/definition.py

```diff
@@ -762,13 +773,21 @@ def metadata_hash(self, audits: t.Dict[str, ModelAudit]) -> str:
 
         for audit_name, audit_args in sorted(self.audits, key=lambda a: a[0]):
             metadata.append(audit_name)
-
+            audit = None
             if audit_name in BUILT_IN_AUDITS:
                 for arg_name, arg_value in audit_args.items():
                     metadata.append(arg_name)
                     metadata.append(gen(arg_value))
+            elif audit_name in self.inline_audits:
+                audit = load_audit(self.inline_audits[audit_name], dialect=self.dialect)
+                if not isinstance(audit, ModelAudit):
+                    audit = None
```

Comment:
How can this ever _not_ be a `ModelAudit`? Is this something we should be asserting instead? The alternative is `StandaloneAudit`, but I don't _think_ we should ever have to deal with these here.